modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -252,3 +279,72 @@ func readyError(status fleet.BundleDeploymentStatus) error {
 
 	return errors.New(msg)
 }
+
+func hasAutoReapply(bd *fleet.BundleDeployment, status *fleet.BundleDeploymentStatus) (ok bool, requeueAfter time.Duration, err error) {
+	requeueAfter = DefaultReapplyInterval
+	if bd.Spec.Options.Resync || bd.Spec.StagedOptions.Resync {
+		if bd.Spec.Options.ResyncPolicy == nil && bd.Spec.StagedOptions.ResyncPolicy == nil {
+			if status.LastApply == nil {
+				return false, requeueAfter, nil
+			}
+			if status.LastApply.Add(DefaultReapplyInterval).Before(time.Now()) {
+				status.LastApply = nil
+				return true, requeueAfter, nil
+			}
+		} else {
+			// use custom resync policy from the template //
+			policy := fleet.ResyncPolicy{}
+
+			if bd.Spec.Options.ResyncPolicy != nil {
+				policy = *bd.Spec.Options.ResyncPolicy
+			}
+
+			if bd.Spec.StagedOptions.ResyncPolicy != nil {
+				policy = *bd.Spec.StagedOptions.ResyncPolicy
+			}
+
+			if policy.ResyncDelay != "" {
+				requeueAfter, err = time.ParseDuration(policy.ResyncDelay)
```

Comment:
If you make `policy.ResyncDelay` as `metav1.Duration`, you don't have to parse at here. It can be done at api level.

Reply:
fixed up.